b2share/modules/records/triggers.py

```diff
@@ -94,6 +90,11 @@ def set_record_owner(record, **kwargs):
     record['owner'] = str(current_user.id)
 
 
+def add_oai_id(record):
```

Comment:
@emanueldima I see that `OAIIDProvider` and `oaiid_fetcher ` [are used in multiple places in invenio-oaiserver](https://github.com/inveniosoftware/invenio-oaiserver/blob/d68f62741c422b34a1438e53b0694570f9ac8e6b/invenio_oaiserver/response.py#L265). Thus, as @jirikuncar said, we need to register the PID. We could add this minter:
```python
def b2share_oaiid_minter(record_uuid, data):
    """Mint record identifiers."""
    pid_value = data.get('_oai', {}).get('id')
    if pid_value is None:
        assert '_deposit' in data and 'id' in data['_deposit']
        pid_value = current_app.config.get('OAISERVER_ID_PREFIX', '') + str(
            data['_deposit']['id']
        )
    provider = OAIIDProvider.create(
        object_type='rec', object_uuid=record_uuid,
        pid_value=str(pid_value)
    )
    data.setdefault('_oai', {})
    data['_oai']['id'] = provider.pid.pid_value
    return provider.pid
```
And use it for every created record, which means: when the deposit is published for the first time. The minter would then be called by the `b2share.modules.deposit.api.Deposit.publish` method (in PR #983)

Note that here I am using the deposit ID as we have the same value for the published record. However at some point we have to check if we can use the same ID for EPIC PIDs too, and if not, decide if we use the internal ID or the EPIC PID as the OAI ID.

Note for later documentation: we have to show organizations deploying B2Share how to configure `OAISERVER_ID_PREFIX`. It must contain the organization namespace as described here: http://www.openarchives.org/OAI/2.0/guidelines-oai-identifier.htm

```
Organizations must choose namespace-identifier values which correspond to a domain-name that they have registered, and are committed to maintaining.
```